examples/01-filter/extract-surface.py

```diff
@@ -0,0 +1,71 @@
+"""
+Extract Surface
+~~~~~~~~~~~~~~~
+
+You can extract the surface of nearly any object within ``pyvista``
+using the ``extract_surface`` filter.
+"""
+
+# sphinx_gallery_thumbnail_number = 2
+
+import numpy as np
+import pyvista as pv
+from vtk import VTK_QUADRATIC_HEXAHEDRON
+
+###############################################################################
+# Create a quadratic cell and extract its surface
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Here we create a single hexahedral cell and then extract its surface
+# to demonstrate how to extract the surface of a UnstructuredGrid.
+
+
+lin_pts = np.array([[-1, -1, -1],  # point 0
+                    [ 1, -1, -1],  # point 1
+                    [ 1,  1, -1],  # point 2
+                    [-1,  1, -1],  # point 3
+                    [-1, -1,  1],  # point 4
+                    [ 1, -1,  1],  # point 5
+                    [ 1,  1,  1],  # point 6
+                    [-1,  1,  1]], np.double)  # point 7
+
+# these are the "midside" points of a quad cell
+quad_pts = np.array([
+    (lin_pts[1] + lin_pts[0])/2.0,
+    (lin_pts[1] + lin_pts[2])/2.0,
+    (lin_pts[2] + lin_pts[3])/2.0,
+    (lin_pts[3] + lin_pts[0])/2.0,
+    (lin_pts[4] + lin_pts[5])/2.0,
+    (lin_pts[5] + lin_pts[6])/2.0,
+    (lin_pts[6] + lin_pts[7])/2.0,
+    (lin_pts[7] + lin_pts[4])/2.0,
+    (lin_pts[0] + lin_pts[4])/2.0,
+    (lin_pts[1] + lin_pts[5])/2.0,
+    (lin_pts[2] + lin_pts[6])/2.0,
+    (lin_pts[3] + lin_pts[7])/2.0], np.double)
+
+# introduce a minor variation to the location of the mid-side points
+quad_pts += np.random.random(quad_pts.shape)*0.3
+pts = np.vstack((lin_pts, quad_pts))
+
+# create the grid
+
+# If you are using vtk>=9, you do not need the offset array
+offset = np.array([0])
+cells = np.asarray(np.hstack((20, np.arange(20))), dtype=np.int64)
+celltypes = np.array([VTK_QUADRATIC_HEXAHEDRON])
```

Comment:
Sadly, yes.  I need to do the same thing I just did for #1213 to the `UnstructuredGrid` class.  We check if all args are `np.ndarray` in:
https://github.com/pyvista/pyvista/blob/07d2fdd0ae655ae3aecf477c84098929effbb4f2/pyvista/core/pointset.py#L593-L594

This will have to be fixed in a separate PR.